gaia-component.js

```diff
@@ -122,6 +54,14 @@ var base = {
         this.setInnerDirAttributes = setInnerDirAttributes.bind(null, this);
         document.addEventListener('dirchanged', this.setInnerDirAttributes);
       }
+      if (document.l10n) {
+        this.localize = document.l10n.translateFragment.bind(
+          document.l10n, this.shadowRoot);
+        document.l10n.ready.then(() => {
+          document.addEventListener('DOMRetranslated', this.localize);
+          this.localize();
+        });
+      }
```

Comment:
Currently this will run for every gaia-component. It might be better for components to opt-in to this feature as so few have localizable shadow content. We could just provide a helper method that select components can call in their `attached()` callback.

```js
attached() {
  this.setupShadowL10n();
}
```

We can do the teardown internally so users don't have to remember to do this.